### PR TITLE
Add Pulse listener

### PR DESCRIPTION
**Pull request by leplatrem**
@magopian r?

### CODE REVIEW

File: pulseworker/main.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+def epoch2iso(timestamp):
+    dt = datetime.datetime.utcfromtimestamp(timestamp)
+    # XXX: Python 3
+    if sys.version_info >= (3, 2):
+        dt = dt.replace(tzinfo=datetime.timezone.utc)
+    return dt.isoformat()
```
Comment (magopian) [inline pulseworker/main.py]: Nothing quite like some good ol' datetime/timezone voodoo <3

File: pulseworker/main.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+
+def pulse2kinto(body, msg, client):
+    # https://github.com/mozilla/pulsetranslator#routing-keys
+    buildinfo = body["payload"]
+
+    skip = (
+        buildinfo["tree"] in ("autoland", "try") or
+        buildinfo["buildtype"] in ("pgo",) or
+        buildinfo["tree"] in ("mozilla-inbound",))
```
Comment (magopian) [inline pulseworker/main.py]: are you expecting some more "skip" in the future for the buildtype and the tree? It feels a bit weird to have this `in` with a single-element list

File: pulseworker/main.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    if skip:
+        logger.debug("Skip routing key '%s'" % body["_meta"]["routing_key"])
+        return  # Do nothing.
+
+    # XXX: no safer way to obtain version?
+    filename = buildinfo["buildurl"].split('/')[-1]
+    version = filename.split('.' + buildinfo["locale"])[0].replace(buildinfo["product"] + '-', '')
```
Comment (magopian) [inline pulseworker/main.py]: Maybe add a comment with an example filename (eg `firefox-55.0a1.en-US.win32.zip`) so it's easier to understand what this piece of code does?

Would it make sense to have a `version_from_filename` helper?

File: pulseworker/main.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        default_server=DEFAULT_SERVER,
+        default_bucket=DEFAULT_BUCKET,
+        default_collection=DEFAULT_COLLECTION)
+
+    args = parser.parse_args(sys.argv[1:])
+
+    cli_utils.setup_logger(logger, args)
+
+    logger.info("Publish at %s/buckets/%s/collections/%s/records" % (
+        args.server, args.bucket, args.collection))
```
Comment (magopian) [inline pulseworker/main.py]: Or you could use `"Publish at {server}/buckets/{bucket}/collections/{collection}".format(**args)` :)

**Comment by leplatrem** [comment]
> I love it, clear, concise, straight to the point, and awesomely well executed, AMAZING!

❤️

File: pulseworker/main.py (hunk 1 above)
Comment (glasserc) [inline pulseworker/main.py]: Personally, I find this kind of code very scary. Could we use `arrow` or something instead?

Comment (leplatrem) [inline pulseworker/main.py]: No need for that, once pulse will be compatible with Python 3 we'll remove the `if`

**Pull request merged**